From 5d2aa6ebaefe0abbbfb45f71787185810d58f1b6 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Fri, 20 Mar 2015 15:05:55 +0800
Subject: [PATCH 0892/1594] MLK-10450: dma: imx-sdma: enable Mega/Fast support
 for i.mx7d

commit 4e1ea64c5d360ebc4f8168c1fcdee314b547bd13 from
git://git.freescale.com/imx/linux-2.6-imx.git

Enable Mega/Fast support for i.mx7d. Need save and restore SDMA registers.

Signed-off-by: Robin Gong <b38343@freescale.com>
---
 drivers/dma/imx-sdma.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index 4cc3b02..c1601ab 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -2082,8 +2082,8 @@ static int sdma_suspend(struct device *dev)
 	struct sdma_engine *sdma = platform_get_drvdata(pdev);
 	int i;
 
-	/* Do nothing if not i.MX6SX */
-	if (sdma->drvdata != &sdma_imx6sx)
+	/* Do nothing if not i.MX6SX or i.MX7D*/
+	if (sdma->drvdata != &sdma_imx6sx && sdma->drvdata != &sdma_imx7d)
 		return 0;
 
 	clk_enable(sdma->clk_ipg);
@@ -2114,8 +2114,8 @@ static int sdma_resume(struct device *dev)
 	struct sdma_engine *sdma = platform_get_drvdata(pdev);
 	int i, ret;
 
-	/* Do nothing if not i.MX6SX */
-	if (sdma->drvdata != &sdma_imx6sx)
+	/* Do nothing if not i.MX6SX or i.MX7D*/
+	if (sdma->drvdata != &sdma_imx6sx && sdma->drvdata != &sdma_imx7d)
 		return 0;
 
 	clk_enable(sdma->clk_ipg);
-- 
1.7.5.4

